lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/driver/evaluator/EvaluatorManager.java

```diff
@@ -487,6 +487,9 @@ public void onResourceLaunch(final ResourceLaunchEvent resourceLaunchEvent) {
       if (this.stateManager.isAllocated()) {
         this.stateManager.setSubmitted();
         this.resourceLaunchHandler.onNext(resourceLaunchEvent);
+      } else if (this.stateManager.isFailedOrKilled()) {
+        LOG.log(Level.WARNING, "Evaluator manager expected" + EvaluatorState.ALLOCATED +
```

Comment:
Shouldn't this generate a `FailedEvaluator`? Or was that generated prior to reaching this?